core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -765,25 +765,29 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
      */
     private fun removeDefensivePacts() {
         val civAtWarWith = otherCiv()
-        civInfo.diplomacy.values.filter { it.diplomaticStatus == DiplomaticStatus.DefensivePact }.forEach {
+        for (diploManager in civInfo.diplomacy.values) {
+            if (diploManager.diplomaticStatus != DiplomaticStatus.DefensivePact) continue
+
             // We already removed the trades and we don't want to remove the flags yet.
-            if (it.otherCiv() != civAtWarWith) {
+            if (diploManager.otherCiv() != civAtWarWith) {
                 // Trades with defensive pact are now invalid
-                val defensivePactOffer = it.trades.firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact } }
-                it.trades.remove(defensivePactOffer)
-                val theirDefensivePactOffer = it.otherCivDiplomacy().trades.firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact } }
-                it.otherCivDiplomacy().trades.remove(theirDefensivePactOffer)
-                it.removeFlag(DiplomacyFlags.DefensivePact)
-                it.otherCivDiplomacy().removeFlag(DiplomacyFlags.DefensivePact)
-                it.diplomaticStatus = DiplomaticStatus.Peace
-                it.otherCivDiplomacy().diplomaticStatus = DiplomaticStatus.Peace
+                val defensivePactOffer = diploManager.trades
+                    .firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact } }
+                diploManager.trades.remove(defensivePactOffer)
+                val theirDefensivePactOffer = diploManager.otherCivDiplomacy().trades
+                    .firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact } }
+                diploManager.otherCivDiplomacy().trades.remove(theirDefensivePactOffer)
+                diploManager.removeFlag(DiplomacyFlags.DefensivePact)
+                diploManager.otherCivDiplomacy().removeFlag(DiplomacyFlags.DefensivePact)
+                diploManager.diplomaticStatus = DiplomaticStatus.Peace
+                diploManager.otherCivDiplomacy().diplomaticStatus = DiplomaticStatus.Peace
```

Comment:
This whole thing is just double code for the same thing.
Either:
A. do `for (manager in listOf(diploManager, diploManager.otherCivDiplomacy())`
B. Create a function `invalidateDefensivePact(diploManager:DiplomacyManager)`

The goal is to increase clarity and ensure we don't miss anything for either side

Reply:
If that is fine, I will continue refactoring this code in a separate branch. I have another branch with defensive pact tests that are very helpful for testing for regression, but they haven't been pushed to the main branch yet.